hazelcast/include/hazelcast/client/impl/metrics/metrics_compressor.h

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cstdint>
+#include <map>
+#include <string>
+#include <vector>
+
+#include <boost/optional.hpp>
+
+#include "hazelcast/client/impl/metrics/metric_descriptor.h"
+#include "hazelcast/client/impl/metrics/metrics_dictionary.h"
+#include "hazelcast/util/byte.h"
+#include "hazelcast/util/export.h"
+
+namespace hazelcast {
+namespace client {
+namespace impl {
+namespace metrics {
+
+class output_buffer {
+public:
+    void write(byte val);
+    void write(int32_t val);
+    void write(int64_t val);
+    void write(double val);
+    void write(const std::string &str);
+    void write(const std::vector<byte> &vec);
+    const std::vector<byte> &content() const;
+    
+private:
+    std::vector<byte> buffer_;
+};
+
+class HAZELCAST_API metrics_compressor
+{
+public:
+    void add_long(const metric_descriptor &descriptor, int64_t value);
+    void add_double(const metric_descriptor &descriptor, double value);
+    std::vector<byte> get_blob();
+
+private:
+    int calculate_descriptor_mask(const metric_descriptor &descriptor);
+    int get_dictionary_id(const boost::optional<std::string> &word);
```

Comment:
```suggestion
    int get_or_put_into_dictionary(const boost::optional<std::string> &word);
```

The method name is not intuitive in the sense that it not only gets but it can also modify the underlying dictionary. A better naming would be easier to understand.

Reply:
getDictionaryId is how it's named in other clients and I don't think `get_or_put_into_dictionary` is more intuitive either.